kube-core/src/dynamic.rs

```diff
@@ -122,7 +122,7 @@ mod test {
 
     #[cfg(feature = "derive")]
     #[tokio::test]
-    #[ignore] // circle has no kubeconfig
+    #[ignore] // needs kubeconfig
```

Comment:
```text
$ cargo check -p kube-core --features=derive
error: none of the selected packages contains these features: derive
```

Also, `needs kubeconfig` is because of `Client::try_default()` (`Config::infer()`), right? We should be able to create a `Config` that's mostly empty, then use that instead.

Reply:
Ah, good spot. The test is definitely in the wrong place (from before kube-core split), so it needs to be moved to root crate (needs core + client + derive).

As a general solution, a blank `Config` ctor could be made, but it should be mared `pub(crate)` only, but then it would only work in `kube-client`.

Reply:
Fixed up the test in root of `kube` for now (also got some mileage out of the new `#[kube(crates..)]` path overrides.